assets/js/components/Dashboard/PermanentConfiguration/Workflows/GlEink/GlEinkWorkflow.tsx

```diff
@@ -66,24 +208,72 @@ const GlEinkWorkflow: ComponentType<WorkflowProps> = ({
         navigate(-1);
       };
       onBack = () => {
+        Object.keys(placesAndScreensToUpdate).forEach((placeId) => {
+          newScreenValidationErrors[placeId] = [];
+        });
+        Object.keys(placesAndScreensToUpdate).forEach((placeId) => {
+          pendingScreenValidationErrors[placeId] = [];
+        });
+
+        dispatch({
+          type: "SET_VALIDATION_ERRORS",
+          newScreenValidationErrors,
+          pendingScreenValidationErrors,
+        });
+        setShowValidationAlert(false);
         setConfigStep(configStep - 1);
       };
       onForward = () => {
-        putPendingScreens(
-          placesAndScreensToUpdate,
-          "gl_eink_v2",
-          configVersion
-        ).then(() => navigate("/pending"));
+        const fieldsWithErrors = validateRequiredFields(
+          placesAndScreensToUpdate
+        );
+
+        if (fieldsWithErrors.size === 0) {
+          putPendingScreens(
+            placesAndScreensToUpdate,
+            "gl_eink_v2",
+            configVersion
+          ).then((response) => {
+            handleGlEinkSubmitResponse(response, fieldsWithErrors);
+          });
+        } else {
+          validateDuplicateScreenIds(placesAndScreensToUpdate);
+          setValidationErrorMessage(generateErrorMessage(fieldsWithErrors));
+          setShowValidationAlert(true);
+          dispatch({
+            type: "SET_VALIDATION_ERRORS",
+            newScreenValidationErrors,
+            pendingScreenValidationErrors,
+          });
+        }
       };
       layout = (
-        <ConfigureScreensWorkflowPage
-          selectedPlaces={places.filter((place) =>
-            selectedPlaces.has(place.id)
+        <div>
```

Comment:
Does this `div` help with styling at all? If not, it can be converted to a fragment.